### PR TITLE
Re Mixup in Cp Space: synthetic Reynolds interpolation for p_re fix

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The p_re OOD split fails because the model sees only **discrete training Reynolds numbers** and must extrapolate to held-out Re values. Standard Mixup in pressure space is wrong because pressure scales as Re² — naive p interpolation creates physically inconsistent samples. However, mixing in **Cp (pressure coefficient) space** creates valid intermediate samples: Cp is Re-invariant, so Cp fields from two same-geometry samples at different Re values can be mixed linearly to produce a geometrically consistent Cp field at an intermediate Re.

Converting back to physical pressure at the interpolated Re value yields a synthetic training sample at a Reynolds number the model has never seen — directly bridging the distribution gap.

This is Mixup (Zhang et al. 2018) applied in the physically correct normalization space for aerodynamic data.

### Step 1: Understand the Reynolds number distribution

Check `cfd_tandemfoil/dataset.py` to find the distinct Re values in the training set. The OOD split holds out certain Re values — these are the distribution gap you're bridging.

### Step 2: Implement `--re_mixup` in the training collation

The implementation should be in the training loop in `train.py` (or a collation wrapper). During training, with probability 0.3 per batch:

```python
def apply_re_mixup(batch, alpha=0.4):
    """
    For each tandem pair in a batch, with probability p=0.3:
    1. Find another sample with the same geometry class but different Re
    2. Compute Cp = p / q_inf for both
    3. Mix Cp fields with lambda ~ Beta(alpha, alpha)
    4. Compute Re_mix = lambda * Re_1 + (1-lambda) * Re_2
    5. Convert back to p = Cp_mix * q_inf(Re_mix)
    6. Return synthetic sample with mixed Re and pressure
    """
    B = len(batch)
    lam = np.random.beta(alpha, alpha)
    
    # For each sample, find a partner with different Re but same config type
    # (single-foil <-> single-foil, tandem <-> tandem)
    for i in range(B):
        if random.random() < 0.3:
            j = find_same_config_different_re(batch, i)
            if j is not None:
                # Get dynamic pressures: q = 0.5 * rho * U^2 where U = Re * nu / c
                q_i = get_dynamic_pressure(batch[i])
                q_j = get_dynamic_pressure(batch[j])
                
                # Mix in Cp space
                cp_i = batch[i]['pressure'] / q_i
                cp_j = batch[j]['pressure'] / q_j
                cp_mix = lam * cp_i + (1 - lam) * cp_j
                
                # Interpolated Re and back-convert to pressure
                re_mix = lam * batch[i]['re'] + (1 - lam) * batch[j]['re']
                q_mix = 0.5 * rho * (re_mix * nu / chord)**2
                
                batch[i]['pressure'] = cp_mix * q_mix
                batch[i]['re'] = re_mix
                
                # Mix velocity fields in U-space (linear in U, not U^2)
                u_i = batch[i]['velocity'] / sqrt(2 * q_i / rho)
                u_j = batch[j]['velocity'] / sqrt(2 * q_j / rho)
                u_mix = lam * u_i + (1 - lam) * u_j
                batch[i]['velocity'] = u_mix * sqrt(2 * q_mix / rho)
    
    return batch
```

Add flags `--re_mixup` and `--re_mixup_alpha 0.4`.

**Important:** Only mix samples of the SAME configuration type (don't mix tandem and single-foil). Use conservative lambda values (Beta(0.4, 0.4) centers around 0.5).

### Step 3: Training runs

```bash
# Seed 42
CUDA_VISIBLE_DEVICES=0,1 python train.py \
  --agent edward --wandb_name "edward/re-mixup-cp-s42" \
  --wandb_group re-mixup-cp --seed 42 \
  --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --slice_num 96 --cosine_T_max 150 --pcgrad_3way \
  --pressure_first --pressure_deep --residual_prediction --surface_refine \
  --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 \
  --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 --wake_angle_feature \
  --vortex_panel_velocity --vortex_panel_scale 0.1 --vortex_panel_n 64 \
  --anp_srf --re_mixup --re_mixup_alpha 0.4

# Seed 73
CUDA_VISIBLE_DEVICES=2,3 python train.py \
  --agent edward --wandb_name "edward/re-mixup-cp-s73" \
  --wandb_group re-mixup-cp --seed 73 \
  --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --slice_num 96 --cosine_T_max 150 --pcgrad_3way \
  --pressure_first --pressure_deep --residual_prediction --surface_refine \
  --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 \
  --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 --wake_angle_feature \
  --vortex_panel_velocity --vortex_panel_scale 0.1 --vortex_panel_n 64 \
  --anp_srf --re_mixup --re_mixup_alpha 0.4
```

If seed 42 shows p_re improvement, run seed 73. If p_in or p_tan degrade, try a lower mixup probability (0.15 instead of 0.3 in the collation).

Note: The Kutta v2 (#2374) work you've been doing showed p_oodc improvements — feel free to also run a combined config (Kutta + Re Mixup) if your seed 42 baseline looks good.

**`--anp_srf` is REQUIRED** — this builds on the ANP baseline.

## Baseline

**Current baseline (PR #2379 — ANP Cross-Foil Decoder, 2-seed avg):**
| Metric | 2-seed avg | Target |
|--------|------------|--------|
| p_in | **3.561** | ≤ 3.561 |
| p_oodc | **3.847** | ≤ 3.847 |
| p_tan | **10.825** | ≤ 10.825 |
| **p_re** | **7.232** | **< 6.229** (recover pre-ANP level) |

p_re is the **primary target**. The Cp-space mixing should create training examples that bridge the OOD Reynolds gap.

Baseline W&B runs: `metqdxaq` (seed 42), `jvfvrs4u` (seed 73)

Reproduce baseline:
```bash
cd cfd_tandemfoil && python train.py --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 --slice_num 96 --cosine_T_max 150 --pcgrad_3way --pressure_first --pressure_deep --residual_prediction --surface_refine --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 --wake_angle_feature --vortex_panel_velocity --vortex_panel_scale 0.1 --vortex_panel_n 64 --anp_srf
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2398 because ADVISOR: Closing — the `--anp_srf` flag does not exist in the codebase. The ANP code from PR #2379 was never committed to the branch (the merge was an empty commit — the code was implemented locally but never pushed). These experiments were designed to fix the ANP p_re regression, but the ANP code itself is missing. You'll receive a new assignment shortly — we're redirecting to bold Round 48 experiments that work on the pre-ANP baseline, plus one student will re-implement the ANP from the W&B config and experiment description..
